samples/materials/texturedLit.mat

```diff
@@ -0,0 +1,60 @@
+material {
+    name : Lit,
+    requires : [
+        uv0
+    ],
+    shadingModel : lit,
+    parameters : [
+        {
+            type : sampler2d,
+            name : albedo
+        },
+        {
+            type : sampler2d,
+            name : roughness
+        },
+        {
+            type : sampler2d,
+            name : metallic
+        },
+        {
+            type : float,
+            name : reflectance
+        },
+        {
+            type : float,
+            name : clearCoat
+        },
+        {
+            type : float,
+            name : clearCoatRoughness
+        },
+        {
+            type : float,
+            name : anisotropy
+        },
+        {
+            type : sampler2d,
+            name : normal
+        },
+        {
+            type : sampler2d,
+            name : ao
+        }
+    ],
+}
+
+fragment {
+    void material(inout MaterialInputs material) {
+        prepareMaterial(material);
+        material.baseColor = texture(materialParams_albedo, getUV0());
+        material.roughness = texture(materialParams_roughness, getUV0()).r;
+        material.metallic = texture(materialParams_metallic, getUV0()).r;
+        material.reflectance = materialParams.reflectance;
```

Comment:
Tip: setting material properties enables new code paths in the shaders, which is expensive. The fewer properties you set, the better.

Reply:
Nice catch, I optimized this a bit with a follow-up CL.